VmArgumentTests AssertionError: mandatory argument not found: -Xjcl:

Failure link
------------

From [an internal build](http://vmfarm.rtp.raleigh.ibm.com/job_output.php?id=108791963):
```
java version "1.8.0_461"
Java(TM) SE Runtime Environment (build 8.0.9.0 - pxz6480sr9-20250603_02(SR9))
IBM J9 VM (build 2.9, JRE 1.8.0 Linux s390x-64-Bit Compressed References 20250612_96338 (JIT enabled, AOT enabled)
OpenJ9   - 9e16360f571
OMR      - 70721030e
IBM      - 8e661b7)
JCL - 20250602_01 based on Oracle jdk8u461-b09
```

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
variation: NoOptions
JVM_OPTIONS:  

FAILED: testArgEncodingInIbmJavaOptions
java.lang.AssertionError: mandatory argument not found: -Xjcl:
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
	at org.testng.AssertJUnit.assertTrue(AssertJUnit.java:24)
	at org.openj9.test.vmArguments.VmArgumentTests.checkArguments(VmArgumentTests.java:1705)
	at org.openj9.test.vmArguments.VmArgumentTests.testArgEncodingInIbmJavaOptions(VmArgumentTests.java:333)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.testng.TestNG.privateMain(TestNG.java:1354)
	at org.testng.TestNG.main(TestNG.java:1323)

   VmArgumentTests
    Tests run: 55, Failures: 47, Skips: 0
```

--- a/test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java
+++ b/test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java
@@ -103,9 +103,7 @@ public class VmArgumentTests {
 	private static final String APPLICATION_NAME = ArgumentDumper.class.getCanonicalName();
 	private static final String HELLO_WORLD = HelloWorld.class.getCanonicalName();
 
-	private static final String[] mandatoryArgumentsJava8;
-
-	private static final String[] mandatoryArguments;
+	private static final List<String> mandatoryArguments;
 	private static final String[] TEST_ARG_LIST = { "-Dtest.option1=testJavaToolOptions1", "-Dtest.option2=testJavaToolOptions2",
 			"-Dtest.option3=testJavaToolOptions3", "-Dtest.option4=testJavaToolOptions4" };
 	private static final String OPTIONS_FILE_SUFFIX = ".test_options_file";
@@ -133,39 +131,26 @@ public class VmArgumentTests {
 	protected static Logger logger = Logger.getLogger(VmArgumentTests.class);
 
 	static {
-		isIBM = System.getProperty("java.vm.vendor").equals("IBM Corporation");
-		mandatoryArgumentsJava8 = new String[] {
-				isIBM ? XOPTIONSFILE : null,
-				"-Xlockword:mode",
-				"-Xjcl:",
-				"-Dcom.ibm.oti.vm.bootstrap.library.path",
-				"-Dsun.boot.library.path",
-				"-Djava.library.path",
-				"-Djava.ext.dirs",
-				DJAVA_HOME,
-				"-Duser.dir",
-				/* "-Djava.runtime.version", doesn't show up if -Xcheck:memory is used */
-				"-Djava.class.path",
-				"-Dsun.java.command",
-				"-Dsun.java.launcher"
-		};
+		isIBM = "IBM Corporation".equals(System.getProperty("java.vm.vendor"));
+		isJava8 = (8 == VersionCheck.major());
 
-		mandatoryArguments = new String[] {
-				isIBM ? XOPTIONSFILE : null,
-				"-Xlockword:mode",
-				"-Xjcl:",
-				"-Dcom.ibm.oti.vm.bootstrap.library.path",
-				"-Dsun.boot.library.path",
-				"-Djava.library.path",
-				DJAVA_HOME,
-				"-Duser.dir",
-				/* "-Djava.runtime.version", doesn't show up if -Xcheck:memory is used */
-				"-Djava.class.path",
-				"-Dsun.java.command",
-				"-Dsun.java.launcher"
-		};
-		int javaMajorVersion = VersionCheck.major();
-		isJava8 = (8 == javaMajorVersion);
+		mandatoryArguments = new ArrayList<>();
+		if (isIBM) {
+			mandatoryArguments.add(XOPTIONSFILE);
+		}
+		mandatoryArguments.add("-Xlockword:mode");
+		mandatoryArguments.add("-Dcom.ibm.oti.vm.bootstrap.library.path");
+		mandatoryArguments.add("-Dsun.boot.library.path");
+		mandatoryArguments.add("-Djava.library.path");
+		if (isJava8) {
+			mandatoryArguments.add("-Djava.ext.dirs");
+		}
+		mandatoryArguments.add(DJAVA_HOME);
+		mandatoryArguments.add("-Duser.dir");
+		mandatoryArguments.add("-Djava.class.path");
+		mandatoryArguments.add("-Dsun.java.command");
+		mandatoryArguments.add("-Dsun.java.launcher");
+
 		vmargsJarFilename = "vmargs.jar";
 	}
 
@@ -1670,12 +1655,8 @@ public class VmArgumentTests {
 	private Map<String, Integer> checkArguments(List<String> actualArguments,
 			String[] optionalArguments) {
 		Map<String, Integer> argPositions = new HashMap<>();
-		String[] mArgs = (isJava8 ? mandatoryArgumentsJava8 : mandatoryArguments).clone();
 		List<String> missing = new ArrayList<>();
-		for (String m : mArgs) {
-			if (null == m) {
-				continue;
-			}
+		for (String m : mandatoryArguments) {
 			boolean found = false;
 			int p = 0;
 			for (String a : actualArguments) {

